acme.go

```diff
@@ -66,6 +67,13 @@ const (
 	boulderBuildIDPath    = "/build"
 )
 
+var (
+	// ErrUnsupportedRSABitLen reports whether an unrecognized RSA key size was used
+	ErrUnsupportedRSABitLen = errors.New("unsupported RSA bit length")
```

Comment:
When one is developing a library there are hundreds of ways people could be
using it. I wouldn't make any assumptions on that regard, and would rather
default to providing types for any error that might happen in order to make
their lives easier. If you still feel strongly about it, I will proceed to
make the error types private.

On Thu, Mar 10, 2016 at 4:45 PM Eric Chiang <notifications@github.com>
wrote:

> In acme.go
> <https://github.com/ericchiang/letsencrypt/pull/19#discussion_r55754836>:
>
> > @@ -66,6 +67,13 @@ const (
> >  	boulderBuildIDPath    = "/build"
> >  )
> >
> > +var (
> > +	// ErrUnsupportedRSABitLen reports whether an unrecognized RSA key size was used
> > +	ErrUnsupportedRSABitLen = errors.New("unsupported RSA bit length")
>
> I'd rather them read the errors and fix their calling code. These aren't
> conditions that seem likely to be resolved programmatically.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ericchiang/letsencrypt/pull/19/files#r55754836>.
>